hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -682,7 +700,7 @@ private synchronized CompletableFuture<Void> handleFlushInternalSynchronized(boo
       LOG.debug("Flushing without data");
     }
     if (putBlockResultFuture != null) {
-      this.lastFlushFuture = watchForCommitAsync(putBlockResultFuture);
+      recordFlushFuture(watchForCommitAsync(putBlockResultFuture));
```

Comment:
https://github.com/apache/ozone/blob/f3b90fb6dfc035e326bd39a289ff0696849b5989/hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java#L428

The one in `writeOnRetry(..)` shown above also needs a similar change.

BTW, since we will always has `recordFlushFuture(watchForCommitAsync(..))` together.  How about combining these two methods?